fleet/pb/fleet.proto

```diff
@@ -6,7 +6,7 @@ option go_package = "fleet/pb";
 service FleetService {
   rpc RetrieveAgent(AgentByIDReq) returns (AgentRes) {}
   rpc RetrieveAgentGroup(AgentGroupByIDReq) returns (AgentGroupRes) {}
-  rpc RetrieveOwnerByChannelID(OwnerByChannelIDReq) returns (OwnerRes) {}
+  rpc RetrieveAgentInfoByChannelID(AgentInfoByChannelIDReq) returns (AgentInfoRes) {}
```

Comment:
so unfortunately I think it's destructive (or at least, not a best practice) with proto buf to directly change definitions like this. we need RPC calls and the messages to be backwards compatible between versions - especially because when upgrades in production happen, there will be several services with mixed versions running and they all need to work together.

so that means we should leave RetrieveOwnerByChannelID and the original messages in this proto definition, and add RetrieveAgentInfoByChannelID and the new messages here as new. the implementation for RetrieveOwnerByChannelID RPC call in code can just be a stub which calls the renamed ViewAgentInfoByChannelIDInternal. the motto is "Don’t Destroy, Deprecate", see notes at:

https://developers.google.com/protocol-buffers/docs/cpptutorial#extending-a-protocol-buffer
https://earthly.dev/blog/backward-and-forward-compatibility/
https://developers.google.com/protocol-buffers/docs/proto3#updating

to be clear, we only have to maintain this backward compatibility in the gRPC/protobuf interface, the renaming in AgentService and AgentRepo are ok.

Reply:
ok, got it, going to read these notes and fix this now